evolvedbinary/prosemirror-lwdita
# Change folder layout: Move unit test folder from `src` to root folder

### Purpose of this PR

This is a preparation for PR #225.  
We need extract all test files from the source folder `src`, as this should contain only the TS modules which are transformed and copied to the `lib` folder.  
Therefore all tests have been moved to the root and the import-, mocharc- and tsconfig-paths have been updated accordingly.

### How to test this PR

1. Run `yarn` to install all dependencies
2. Run `yarn coverage` to run the unit tests and get the coverage output in the console
3. **Note:** `yarn build` will currently fail on the main branch. You can work around this issue in order to test the changes done in `tsconfig.ts` by adding line `"skipLibCheck": true` to file `tsconfig.ts`.

The current, expected coverage should now output
<img width="917" alt="Screenshot 2024-03-21 at 16 37 43" src="https://github.com/evolvedbinary/prosemirror-jdita/assets/1157669/0ff133ee-d41e-4a4f-a8fb-534fee290a18">
